docs/src/pages/components/data-grid/filtering/CustomFilterPanel.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react';
+import {
+  DataGrid,
+  GridToolbarContainer,
+  GridToolbarFilterButton,
+} from '@mui/x-data-grid';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+const VISIBLE_FIELDS = ['name', 'rating', 'country', 'dateCreated', 'isAdmin'];
+
+export default function CustomFilterPanel() {
+  const { data } = useDemoData({
+    dataSet: 'Employee',
+    visibleFields: VISIBLE_FIELDS,
+    rowLength: 100,
+  });
+
+  const [filterButtonEl, setFilterButtonEl] =
+    React.useState<HTMLButtonElement | null>(null);
+
+  const CustomToolbar = React.useCallback(() => {
+    return (
+      <GridToolbarContainer>
+        <GridToolbarFilterButton ref={setFilterButtonEl} />
+      </GridToolbarContainer>
+    );
+  }, []);
```

Comment:
I think it would be more realistic to define the `CustomToolbar` outside of the `CustomFilterPanel`. Because in a real-world project, the `CustomToolbar` will be imported from another file.

Reply:
I copied Matheus example
I think he did it that way to avoid a React Context
But I agree, now that I am putting it in the doc, we should make it more realistic :+1:

Reply:
Done :+1:

Reply:
Alexandre is right. In this case `componentsProps.toolbar` could be used to pass the prop to the custom toolbar.

```tsx
const CustomToolbar = ({ setFilterButtonEl }) => {
  return (
    <GridToolbarContainer>
      <GridToolbarFilterButton ref={setFilterButtonEl} />
    </GridToolbarContainer>
  );
};

export default function CustomFilterPanel() {
  const [filterButtonEl, setFilterButtonEl] = React.useState(null);

  return (
    <div style={{ height: 400, width: '100%' }}>
      <DataGrid
        columns={columns}
        rows={rows}
        components={{ Toolbar: CustomToolbar }}
        componentsProps={{
          panel: { anchorEl: filterButtonEl },
          toolbar: { setFilterButtonEl },
        }}
      />
    </div>
  );
}

Reply:
Good point